project_weather_app/main.py

```diff
@@ -22,20 +42,25 @@ def main():
         print("Database: ", MOCKED_DB_FILE)
         db_file = MOCKED_DB_FILE
     else:
-        raise("invalid mode")
+        raise e.ModeError("Mode error.", MODE,2)
+    print("Mode: ", MODE)
 
     # create database to store weather data
     db = WeatherDatabase(db_file)
 
+    ### Drop database table (for testing purpose)
+    # db.drop()
+    # exit(0)
+
     # setup the data_service and data_handler
     data_service = DataServiceFactory(DATA_SOURCE, database=db, mode=MODE).create()    
     data_handler = DataHandler(data_service)
 
     # handle the data for the given location
-    location = Location(longitude=LONGITUDE, latitude=LATITUDE)
     data_handler.execute(location)
 
     print("Program finished.")
+    exit(0)
```

Comment:
you don't really need the `exit(0)`, since the python program will exit gracefully anyway once your main file's last line of code is executed without any unhandled exception being raised